### DISCUSSION TITLE
Tauri csp setting - different for "new" vs "add to existing" project

### DISCUSSION THREAD

**Discussion by nokola** [tauri-apps/tauri#2992] [Q&A] [answered]
I noticed a difference in "csp" setting when I followed the [add tauri to new project](https://tauri.studio/en/docs/usage/development/integration#1-start-a-new-tauri-project) vs [add to existing project](https://tauri.studio/en/docs/usage/development/integration#1-add-tauri-to-existing-project) docs

New project has this in tauri.config.js:
```
      "csp": "default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'"
```

Existing project docs end up with this:
```
      "csp": "default-src blob: data: filesystem: ws: http: https: 'unsafe-eval' 'unsafe-inline'"
```

I'm not sure what the implication is. I'll keep the setting from the "new project" in my existing project. Is the difference intentional or just a bug?

**Comment by FabianLars** [comment, marked as answer]
If we're on the same base as in your recent issue, then the problem is an outdated cli, because yarn 3 resolves `^1.0.0-beta.10` to `1.0.0-beta-rc.4`. So the fix is still the same -> replace the `^` with and `=` and run `yarn install` again.
Because we don't handle yarn 3 very well atm, you need to manually unplug the cli package if you use pnp so it can download the actual binary.
Edit: So the "correct" one is the first one, but either way they are meant to be edited according to your needs to further secure your app.

**Reply by nokola** [reply]
Thanks, got it!
